docs: Update Applications Management API documentation (#3139)
bonitasoft/bonita-doc: modules/applications/pages/ui-builder/rest-api.adoc

* Deploy initial applications
* Update applications: add, update, and remove applications
* Re-deploy applications: ensuring resilience (for example in case of a Kubernetes pod crash)

> * Export all applications for store files in a Bonita Studio project

Reviewer: ```suggestion
* Export all applications for storing files in a Bonita Studio project
```

The `X-Bonita-API-Token` and `JSESSIONID` cookies are required for these APIs (in the Headers).
====

== Overview

> The Applications Management API provides endpoints to manage applications in the BonitAppsmith workspace. These endpoints allow you to list, import, export, and delete applications.

Reviewer: ```suggestion
The Applications Management API provides endpoints to manage applications in the UIB workspace. These endpoints allow you to list, import, export, and delete applications.
```

**Base URL**: `/uib/api/v1/applications`

**Common Requirements**:

* All endpoints require authentication via session cookies

> * CSRF protection via `x-requested-by: Appsmith` header is required

Reviewer: ```suggestion
* CSRF protection requires header `x-requested-by: Appsmith` to be present
```

  --header 'x-requested-by: Appsmith' \
  --header 'Cookie: X-Bonita-API-Token=abc123; JSESSIONID=def456'
----

**Response:**
[source,json]
----
{
  "responseMeta": {
    "status": 200,
    "success": true
  },
  "data": [
    {"id": "64a1b2c3d4e5f6", "name": "Dashboard"},
    {"id": "74b2c3d4e5f6a7", "name": "Admin Panel"}
  ],
  "errorDisplay": null
}
----

==== Example 2: List applications (full details)

[source,bash]
----
curl -X GET 'http://localhost/uib/api/v1/applications/?full=true' \
  --header 'x-requested-by: Appsmith' \
  --header 'Cookie: X-Bonita-API-Token=abc123; JSESSIONID=def456'
----

Returns complete application objects with all fields.

==== Example 3: Empty workspace

[source,bash]
----
curl -X GET 'http://localhost/uib/api/v1/applications/' \
  --header 'x-requested-by: Appsmith' \
  --header 'Cookie: X-Bonita-API-Token=abc123; JSESSIONID=def456'
----

**Response:**
[source,json]
----
{
  "responseMeta": {
    "status": 204,
    "success": true
  },
  "data": [],
  "errorDisplay": "No applications found"
}
----

=== Behavior Details

* Retrieves applications from the default workspace only
* Returns applications in the order they are stored

> * The `full` parameter allows control over response payload size

Reviewer: correct but, just an (IA) suggestion to make it easier to read 
```suggestion
* The `full` parameter allows to control the size of the response payload
```

* **Name Conflicts**: If an application with the same `slug` exists, the import fails
* **Validation**: The JSON must be a valid Appsmith application descriptor

[[import-multiple-applications]]
== Import Multiple Applications

Import multiple applications from a ZIP archive containing JSON descriptor files.

=== Request

==== HTTP Method
`POST`

==== URL
`/uib/api/v1/applications/import-bulk`

==== Headers

[cols="1,1,3"]
|===
|Header |Required |Description

|`x-requested-by`
|Yes
|Must be set to `Appsmith`

|`Cookie`
|Yes
|Must include valid session cookies: `X-Bonita-API-Token` and `JSESSIONID`

|`Content-Type`
|Yes
|Must be `multipart/form-data`
|===

==== Request Body

[cols="1,1,1,3"]
|===
|Field |Type |Required |Description

|`file`
|File
|Yes
|The ZIP archive containing JSON application descriptors. Must have `.zip` extension.
|===

=== Response

==== Content Type
`application/json`

==== Response Status Codes

[cols="1,1,3"]
|===
|Status Code |Status |Description

|200
|OK
|All applications imported successfully.

|206
|PARTIAL_CONTENT
|Some applications were imported, but not all. Check response for details.

|400
|BAD_REQUEST

> |Invalid file (not a file part, not a .zip file, or no applications imported from the archive).

Reviewer: ```suggestion
|Invalid file (not a file part, not a .zip file, or no valid applications found in the archive).
```

INFO  - Number of applications exported: 5
INFO  - Export finished, streaming zip to client
```

=== Troubleshooting with Logs

Check server logs to diagnose:

* Individual application failures in bulk operations
* Performance bottlenecks
* File system issues
* Database connection problems
* Authentication/authorization failures

== API Version and Compatibility

* **Current Version**: v1
* **Base Path**: `/uib/api/v1/applications`
* **Stability**: Production-ready
* **Backward Compatibility**: Changes to v1 endpoints maintain backward compatibility

=== Future Considerations

Potential future enhancements:

* Asynchronous processing for large operations
* Progress tracking for long-running exports/imports
* Selective export (by application IDs)
* Scheduled automated backups
* Workspace selection parameter
* Dry-run mode for import operations
* Diff comparison between environments

== Additional Resources

* **Export Format Specification**: Appsmith `ArtifactExchangeJson` format
* **Import Service Documentation**: ApplicationDescriptorHelper implementation
* **File System Configuration**: `bonitaProperties.applicationFolderPath`

> * **Error Codes Reference**: Standard HTTP status codes

Reviewer: is all this really needed ?

Author: No, i forgot to remove this part !

> For support or questions, consult the BonitAppsmith documentation or contact your system administrator.

Reviewer: ```suggestion
For support or questions, consult the UIB documentation or contact your system administrator.
```
